python/setup.py

```diff
@@ -337,9 +338,10 @@ def _run_cmake(self, pyarrow_cpp_home):
             static_lib_option = ''
 
             cmake_options = [
-                '-DPYTHON_EXECUTABLE=%s' % sys.executable,
-                '-DPython3_EXECUTABLE=%s' % sys.executable,
+                '-DPYTHON_EXECUTABLE=' + str(sys.executable),
+                '-DPython3_EXECUTABLE=' + str(sys.executable),
```

Comment:
Nit: it should already be a string:
```suggestion
                '-DPYTHON_EXECUTABLE=' + sys.executable,
                '-DPython3_EXECUTABLE=' + sys.executable,
```

Reply:
Hmm, I see the same idiom is used above actually...

Reply:
Yeah, I am OK with removing it everywhere.